Content.Server/Doors/Systems/AirlockSystem.cs

```diff
@@ -109,5 +112,43 @@ private void OnDoorPry(EntityUid uid, AirlockComponent component, BeforeDoorPryE
                 args.Cancel();
             }
         }
+
+        private void OnRangedInteract(EntityUid uid, AirlockComponent component, RangedInteractEvent args)
+        {
+            args.Handled = true;
+            var doorComponent = EntityManager.GetComponent<ServerDoorComponent>(component.Owner);
+            
+            // If it isn't a door remote we don't use it
+            if(!EntityManager.TryGetComponent<DoorRemoteComponent?>(args.UsedUid, out var remoteComponent))
+            {
+                args.Handled = false;
+                return;
+            }
```

Comment:
Shouldn't this be handled on the doorremote itself?

Reply:
RangedInteractEvent is raised on the target and I don't know if there is a way to interact out-of-range in another way.
https://github.com/space-wizards/space-station-14/blob/cea1b21832eb3181f289a33ab85a8d7a614486c7/Content.Server/Interaction/InteractionSystem.cs#L275
```
RaiseLocalEvent(target.Value, rangedMsg);
```

Reply:
All the different interaction types are a confusing mess, but I think `AfterInteractEvent` should be raised directed at the used item if no ranged interaction occurred, so you should be able to use that.

Reply:
You would appear to be correct.
I will try to move the functionality to remote.

Reply:
Unfortunately `AfterInteractEvent`does not suit my needs as `InteractUsingRanged` passes target as `null` to `InteractDoAfter` if it is not in range.
https://github.com/space-wizards/space-station-14/blob/cea1b21832eb3181f289a33ab85a8d7a614486c7/Content.Server/Interaction/InteractionSystem.cs#L293

Reply:
Yeah I think this is the first time that (non-combat/shooting) ranged interactions are actually being used.

I guess you can leave it as is, but if you wanted to (in this, or a future PR) feel free to just redo how to ranged stuff works so that you could actually use an event directed at the door remote

Reply:
It has been moved from AirlockSystem to DoorRemoteSystem.